src/Model/Behavior/DuplicatableBehavior.php

```diff
@@ -329,4 +332,34 @@ protected function _doAction($action, EntityInterface $entity, $prop, $value = n
                 break;
         }
     }
+
+    /**
+     * Removes the primary key from the entity being duplicated.  If the entity has a composite key, you must
+     * specify which field we wish to preserve
+     *
+     * @param EntityInterface $entity
+     * @param $object
+     * @throws \Exception
+     */
+    protected function removePrimaryKey(EntityInterface $entity, $object)
+    {
+        if (is_array($object->getPrimaryKey())) {
+
+            $compositeKey = $object->getPrimaryKey();
```

Comment:
Make this assignment the first statement (with name $primaryKey) and avoid calling method again.

Reply:
Done